flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -359,21 +359,14 @@ public Task(
 
 		final String taskNameWithSubtaskAndId = taskNameWithSubtask + " (" + executionId + ')';
 
-		// add metrics for buffers
-		final MetricGroup buffersGroup = metrics.getIOMetricGroup().addGroup("buffers");
-
-		// similar to MetricUtils.instantiateNetworkMetrics() but inside this IOMetricGroup
-		final MetricGroup networkGroup = metrics.getIOMetricGroup().addGroup("Network");
-		final MetricGroup outputGroup = networkGroup.addGroup("Output");
-		final MetricGroup inputGroup = networkGroup.addGroup("Input");
+		MetricGroup taskShuffleMetricGroup = networkEnvironment.createTaskMetricGroup(metrics.getIOMetricGroup());
```

Comment:
Does it mean `ShuffleEnvironment` interface need provide the method `createTaskMetricGroup`?

Reply:
My idea was to expose `TaskShuffleContext ShuffleEnvironment.createTaskShuffleContext(taskName, execId, parentMetricGroup)` in `ShuffleEnvironment `. `TaskShuffleContext` would contain  taskName, execId and created in/output metric group as well. `createPartition/Gate` methods would take the `TaskShuffleContext`. This way `ShuffleEnvironment ` can create its own metric group once and other task common parameters are encapsulated.